### PR TITLE
Add namespaced e2e tests

### LINKED ISSUE #2408: Enhance E2E testing to include ns scoped resources
https://github.com/kubernetes-incubator/service-catalog/pull/2370 fixed a core issue around creating a namespace scoped ServiceInstance (a prior PR introduced a bug which prevented creating ns'd ServiceInstances).  It also added a unit test to catch the issue, but overall it points out none of our existing E2E tests exercise namespace scoped resources.  It would be smart to add some.

### PR DESCRIPTION
**Pull request by jberkhahn**
- add e2e tests that use namespace resources
- change build-e2e make target to compile on host os/arch
- refactor test/util helper methods to work on both cluster/ns resources

This PR is a 
 - [x] Feature Implementation
 - [ ] Bug Fix
 - [ ] Documentation

Fixes #2408 

Merge Checklist:
 - [ ] New feature 
   - [ ] Tests
   - [ ] Documentation
 - [ ] SVCat CLI flag
 - [ ] Server Flag for config
   - [ ] Chart changes
   - [ ] removing a flag by marking deprecated and hiding to avoid
         breaking the chart release and existing clients who provide a
         flag that will get an error when they try to update

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -175,7 +175,7 @@ $(BINDIR)/%-gen: $$(shell find vendor/k8s.io/code-generator/cmd/$$*-gen vendor/k
 
 .PHONY: $(BINDIR)/e2e.test
 $(BINDIR)/e2e.test: .init
-	$(DOCKER_CMD) go test -c -o $@ $(SC_PKG)/test/e2e
+	$(DOCKER_CMD) env GOOS=$(CLIENT_PLATFORM) GOARCH=$(ARCH) go test -c -o $@ $(SC_PKG)/test/e2e
```
Comment (jberkhahn) [inline Makefile]: @MHBauer could you run this make task on your machine to make sure it works somewhere other than my machine?

File: test/util/util.go [hunk 2]
```diff
@@ -32,29 +32,38 @@ import (
...
 	scfeatures "github.com/kubernetes-incubator/service-catalog/pkg/features"
+	servicecatalog "github.com/kubernetes-incubator/service-catalog/pkg/svcat/service-catalog"
 	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 )
 
 // WaitForBrokerCondition waits for the status of the named broker to contain
-// a condition whose type and status matches the supplied one.
-func WaitForBrokerCondition(client v1beta1servicecatalog.ServicecatalogV1beta1Interface, name string, condition v1beta1.ServiceBrokerCondition) error {
+// a condition whose type and status matches the supplied onei. Checks for a
```
Comment (jboyd01) [inline test/util/util.go]: nit: one vs onei

Comment (jberkhahn) [reply test/util/util.go]: fixed

**Comment by MHBauer** [comment]
onei
/lgtm
